File: Fuentes/Principal/Entidades/Usuarios.cs
using System;
using System.Collections.Generic;
using System.Data.SqlClient;
using System.Linq;
using System.Text;

namespace Entidades
{
    public class Usuarios
    {
         
        private int id;
        private string nombre;
        private string contrasena;
        private int nivel;
        private bool accesoTotal; 
         
        public int Id
        {
            get { return id; }
            set { id = value; }
        }
        public string Nombre
        {
            get { return nombre; }
            set { nombre = value; }
        }
        public string Contrasena
        {
            get { return contrasena; }
            set { contrasena = value; }
        }
        public int Nivel
        {
            get { return nivel; }
            set { nivel = value; }
        }
        public bool AccesoTotal
        {
            get { return accesoTotal; }
            set { accesoTotal = value; }
        } 

        public void Guardar()
        {

            try
            {
                SqlCommand comando = new SqlCommand();
                comando.Connection = BaseDatos.conexionCatalogo;
                comando.CommandText = "INSERT INTO Usuarios VALUES (@id, @nombre, @contrasena, @nivel, @accesoTotal)"; 
                comando.Parameters.AddWithValue("@id", this.Id);
                comando.Parameters.AddWithValue("@nombre", this.Nombre);
                comando.Parameters.AddWithValue("@contrasena", this.Contrasena);
                comando.Parameters.AddWithValue("@nivel", this.Nivel);
                comando.Parameters.AddWithValue("@accesoTotal", this.AccesoTotal); 
                BaseDatos.conexionCatalogo.Open();
                comando.ExecuteNonQuery();
                BaseDatos.conexionCatalogo.Close();
            }
            catch (Exception ex)
            {
                throw ex;
            }
            finally
            {
                BaseDatos.conexionCatalogo.Close();
            }

        }

        public void Editar()
        {

            try
            {
                SqlCommand comando = new SqlCommand();
                comando.Connection = BaseDatos.conexionCatalogo;
                comando.CommandText = "UPDATE Usuarios SET Nombre=@nombre, Contrasena=@contrasena, Nivel=@nivel, AccesoTotal=@accesoTotal WHERE Id=@id"; 
                comando.Parameters.AddWithValue("@id", this.Id);
                comando.Parameters.AddWithValue("@nombre", this.Nombre);
                comando.Parameters.AddWithValue("@contrasena", this.Contrasena);
                comando.Parameters.AddWithValue("@nivel", this.Nivel);
                comando.Parameters.AddWithValue("@accesoTotal", this.AccesoTotal);
                BaseDatos.conexionCatalogo.Open();
                comando.ExecuteNonQuery();
                BaseDatos.conexionCatalogo.Close();
            }
            catch (Exception ex)
            {
                throw ex;
            }
            finally
            {
                BaseDatos.conexionCatalogo.Close();
            }

        }

        public void Eliminar()
        {

            try
            {
                SqlCommand comando = new SqlCommand();
                comando.Connection = BaseDatos.conexionCatalogo;
                string condiciones = string.Empty;
                if (this.Id > 0)
                {
                    condiciones += " WHERE Id=@id";
                }
                comando.CommandText = "DELETE FROM Usuarios";
                comando.Parameters.AddWithValue("@id", this.Id);
                BaseDatos.conexionCatalogo.Open();
                comando.ExecuteNonQuery();
                BaseDatos.conexionCatalogo.Close();
            }
            catch (Exception ex)
            {
                throw ex;
            }
            finally
            {
                BaseDatos.conexionCatalogo.Close();
            }

        }
         
        public List<Usuarios> ObtenerListadoPorNombre()
        {

            try
            {
                List<Usuarios> lista = new List<Usuarios>();
                SqlCommand comando = new SqlCommand();
                comando.Connection = BaseDatos.conexionCatalogo;
                comando.CommandText = "SELECT * FROM Usuarios WHERE Nombre=@nombre"; 
                comando.Parameters.AddWithValue("@nombre", this.Nombre);
                BaseDatos.conexionCatalogo.Open();
                SqlDataReader lectorDatos = comando.ExecuteReader();
                Usuarios usuarios = default(Usuarios);
                while (lectorDatos.Read())
                {
                    usuarios = new Usuarios(); 
                    usuarios.id = Convert.ToInt32(lectorDatos["id"].ToString());
                    usuarios.nombre = lectorDatos["nombre"].ToString();
                    usuarios.contrasena = lectorDatos["contrasena"].ToString();
                    usuarios.nivel = Convert.ToInt32(lectorDatos["nivel"].ToString());
                    usuarios.accesoTotal = Convert.ToBoolean(lectorDatos["accesoTotal"].ToString()); 
                    lista.Add(usuarios);
                }
                BaseDatos.conexionCatalogo.Close();
                return lista;
            }
            catch (Exception ex)
            {
                throw ex;
            }
            finally
            {
                BaseDatos.conexionCatalogo.Close();
            }

        }

        public bool ValidarPorId()
        {

            try
            {
                bool resultado = false;
                SqlCommand comando = new SqlCommand();
                comando.Connection = BaseDatos.conexionCatalogo;
                comando.CommandText = "SELECT * FROM Usuarios WHERE Id=@id"; 
                comando.Parameters.AddWithValue("@id", this.Id);
                BaseDatos.conexionCatalogo.Open();
                SqlDataReader lectorDatos = comando.ExecuteReader();
                if (lectorDatos.HasRows)
                {
                    resultado = true;
                }
                else
                {
                    resultado = false;
                }                
                BaseDatos.conexionCatalogo.Close();
                return resultado;
            }
            catch (Exception ex)
            {
                throw ex;
            }
            finally
            {
                BaseDatos.conexionCatalogo.Close();
            }

        }

        public List<Usuarios> ObtenerListado()
        {

            try
            {
                List<Usuarios> lista = new List<Usuarios>();
                SqlCommand comando = new SqlCommand();
                comando.Connection = BaseDatos.conexionCatalogo;
                comando.CommandText = "SELECT * FROM Usuarios";
                BaseDatos.conexionCatalogo.Open();
                SqlDataReader lectorDatos = comando.ExecuteReader();
                Usuarios usuarios;
                while (lectorDatos.Read())
                {
                    usuarios = new Usuarios();
                    usuarios.Id = Convert.ToInt32(lectorDatos["id"].ToString());
                    usuarios.Nombre = lectorDatos["nombre"].ToString();
                    usuarios.Contrasena = lectorDatos["contrasena"].ToString();
                    usuarios.Nivel = Convert.ToInt32(lectorDatos["nivel"].ToString());
                    usuarios.AccesoTotal = Convert.ToBoolean(lectorDatos["accesoTotal"].ToString()); 
                    lista.Add(usuarios);
                }
                BaseDatos.conexionCatalogo.Close();
                return lista;
            }
            catch (Exception ex)
            {
                throw ex;
            }
            finally
            {
                BaseDatos.conexionCatalogo.Close();
            }

        }

        public List<Usuarios> ObtenerListadoPorId()
        {

            try
            {
                List<Usuarios> lista = new List<Usuarios>();
                SqlCommand comando = new SqlCommand();
                comando.Connection = BaseDatos.conexionCatalogo;
                comando.CommandText = "SELECT * FROM Usuarios WHERE Id=@id";
                comando.Parameters.AddWithValue("@id", this.id);
                BaseDatos.conexionCatalogo.Open();
                SqlDataReader lectorDatos = comando.ExecuteReader();
                Usuarios usuarios = default(Usuarios);
                while (lectorDatos.Read())
                {
                    usuarios = new Usuarios();
                    usuarios.id = Convert.ToInt32(lectorDatos["id"].ToString());
                    usuarios.nombre = lectorDatos["nombre"].ToString();
                    usuarios.contrasena = lectorDatos["contrasena"].ToString();
                    usuarios.nivel = Convert.ToInt32(lectorDatos["nivel"].ToString());
                    usuarios.accesoTotal = Convert.ToBoolean(lectorDatos["accesoTotal"].ToString()); 
                    lista.Add(usuarios);
                }
                BaseDatos.conexionCatalogo.Close();
                return lista;
            }
            catch (Exception ex)
            {
                throw ex;
            }
            finally
            {
                BaseDatos.conexionCatalogo.Close();
            }

        }

    }
}

File: Fuentes/Catalogos (Obsoleto)/Entidades/Usuarios.vb
Imports System.Collections.Generic
Imports System.Data.SqlClient
Imports System.Linq
Imports System.Text
Imports System.ComponentModel

Public Class Usuarios
     
    Private id As Integer
    Private nombre As String
    Private contrasena As String
    Private nivel As Integer
    Private accesoTotal As Boolean 
     
    Public Property EId() As Integer
        Get
            Return id
        End Get
        Set(value As Integer)
            id = value
        End Set
    End Property
    Public Property ENombre() As String
        Get
            Return nombre
        End Get
        Set(value As String)
            nombre = value
        End Set
    End Property
    Public Property EContrasena() As String
        Get
            Return contrasena
        End Get
        Set(value As String)
            contrasena = value
        End Set
    End Property
    Public Property ENivel() As Integer
        Get
            Return nivel
        End Get
        Set(value As Integer)
            nivel = value
        End Set
    End Property
    Public Property EAccesoTotal() As Boolean
        Get
            Return accesoTotal
        End Get
        Set(value As Boolean)
            accesoTotal = value
        End Set
    End Property 
     
    Public Function ValidarPorId() As Boolean

        Try
            Dim resultado As Boolean = False
            Dim comando As New SqlCommand()
            comando.Connection = BaseDatos.conexionCatalogo
            comando.CommandText = "SELECT * FROM Usuarios WHERE Id=@id"
            comando.Parameters.AddWithValue("@id", Me.Id)
            BaseDatos.conexionCatalogo.Open()
            Dim dataReader As SqlDataReader = comando.ExecuteReader()
            If (dataReader.HasRows) Then
                resultado = True
            Else
                resultado = False
            End If
            BaseDatos.conexionCatalogo.Close()
            Return resultado
        Catch ex As Exception
            Throw ex
        Finally
            BaseDatos.conexionCatalogo.Close()
        End Try

    End Function

    Public Function ObtenerListado() As List(Of Usuarios)

        Try
            Dim lista As New List(Of Usuarios)
            Dim comando As New SqlCommand()
            comando.Connection = BaseDatos.conexionCatalogo
            Dim condicion As String = String.Empty
            If (Me.EId > 0) Then
                condicion &= " WHERE Id=@id"
            End If
            comando.CommandText = "SELECT * FROM Usuarios " & condicion
            comando.Parameters.AddWithValue("@id", Me.id)
            BaseDatos.conexionCatalogo.Open()
            Dim dataReader As SqlDataReader = comando.ExecuteReader()
            Dim tabla As Usuarios
            While dataReader.Read()
                tabla = New Usuarios()
                tabla.id = Convert.ToInt32(dataReader("Id").ToString())
                tabla.nombre = dataReader("Nombre").ToString()
                tabla.contrasena = dataReader("Contrasena").ToString()
                tabla.nivel = Convert.ToInt32(dataReader("Nivel").ToString())
                tabla.accesoTotal = Convert.ToBoolean(dataReader("AccesoTotal").ToString())
                lista.Add(tabla)
            End While
            BaseDatos.conexionCatalogo.Close()
            Return lista
        Catch ex As Exception
            Throw ex
        Finally
            BaseDatos.conexionCatalogo.Close()
        End Try

    End Function

    Public Function ObtenerListadoReporte() As DataTable

        Try
            Dim datos As New DataTable
            Dim comando As New SqlCommand()
            comando.Connection = BaseDatos.conexionCatalogo
            comando.CommandText = "SELECT Id, Nombre FROM Usuarios ORDER BY Id"
            BaseDatos.conexionCatalogo.Open()
            Dim dataReader As SqlDataReader
            dataReader = comando.ExecuteReader()
            datos.Load(dataReader)
            BaseDatos.conexionCatalogo.Close()
            Return datos
        Catch ex As Exception
            Throw ex
        Finally
            BaseDatos.conexionCatalogo.Close()
        End Try

    End Function

    Public Function ValidarActividadPorId() As Boolean

        Try
            Dim resultado As Boolean = False
            Dim comando As New SqlCommand()
            comando.Connection = BaseDatos.conexionAgenda
            comando.CommandText = "SELECT * FROM Actividades WHERE IdUsuario=@id OR IdUsuarioDestino=@id"
            comando.Parameters.AddWithValue("@id", Me.Id)
            BaseDatos.conexionAgenda.Open()
            Dim dataReader As SqlDataReader = comando.ExecuteReader()
            If dataReader.HasRows Then
                resultado = True
            Else
                resultado = False
            End If
            BaseDatos.conexionAgenda.Close()
            Return resultado
        Catch ex As Exception
            Throw ex
        Finally
            BaseDatos.conexionAgenda.Close()
        End Try

    End Function

End Class
